### PR TITLE
fix: Latest configuration for `devcontainer.json` and Dockerfile

### PR DESCRIPTION
**Pull request by mdsanima**
#### Description

_Please explain the changes you made here._

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [ ] Verified the license of any newly added font, glyph, or glyph set

#### What does this Pull Request (PR) do?

Updating the `.devcontainer` configuration to latest version.

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,6 @@
-ARG VARIANT="3"
-FROM mcr.microsoft.com/vscode/devcontainers/python:0-${VARIANT}
+ARG VARIANT=3
+FROM mcr.microsoft.com/vscode/devcontainers/python:${VARIANT}
```
Comment (Finii) [inline .devcontainer/Dockerfile:2]: Just for documentation:

See https://github.com/microsoft/vscode-dev-containers/blob/main/containers/python-3/README.md#configuration

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -1,32 +1,35 @@
...
-// https://github.com/microsoft/vscode-dev-containers/tree/v0.122.1/containers/python-3
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/python-3
 {
     "name": "Python 3",
     "build": {
         "dockerfile": "Dockerfile",
         "context": "..",
-        "args": { "VARIANT": "3" }
+        "args": {
+            "VARIANT": "3"
```
Comment (Finii) [inline .devcontainer/devcontainer.json:9]: I believe we do not need Node?

```suggestion
            "VARIANT": "3"
            "NODE_VERSION": "none"
```

**Review by Finii — COMMENTED** [review]
Thanks for the PR!

The devcontainer has been introduced silently with
* #497 

and no explanation given why we possibly want that.

Maybe you can explain what the advantages are or the use-cases, maybe adding documentation. To be honest I was on the brink of removing it some months ago for uselessness and non-documentation.

File: .devcontainer/Dockerfile [hunk 3]
```diff
@@ -1,7 +1,6 @@
...
 
-RUN apt-get update \
-    && export DEBIAN_FRONTEND=noninteractive \
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
```
[the remark this answers is not in the document]
Comment (mdsanima) [reply .devcontainer/Dockerfile:4]: This change is from [official documentation](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/python-3/.devcontainer/Dockerfile#L15).

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (mdsanima) [reply .devcontainer/Dockerfile:2]: This changes also is from [official documentation](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/python-3/.devcontainer/Dockerfile#L2C1-L2C1) but we put the latest version of Python.

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (mdsanima) [reply .devcontainer/devcontainer.json]: Now I added this option too.

**Comment by Finii** [comment]
> As for documentation, I can write it, can you tell me where? Should this be in the READMD.md file or maybe in some other file?

That would be great! Just a `./devcontainer/README.md` maybe is enough?

File: .devcontainer/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (Finii) [reply .devcontainer/Dockerfile:4]: I'd rather keep it one command per line as before. In principle it is the same, but how they write it it is misleading - the DEBIAN_FRONTEND envvar is used by the next command (apt-get install) and has nothing to do with the update.

**Comment by Finii** [comment]
@allcontributors please add @mdsanima  for code

File: .devcontainer/Dockerfile (hunk 3 above)
[the remark this answers is not in the document]
Comment (mdsanima) [reply .devcontainer/Dockerfile:4]: Ok, I rebase this!

**Comment by Finii** [comment]
Personally I believe translations age not gracefully, i.e. get out of sync more or less quickly.

Thanks for adding the explanation/readme!

I will pull this in the next minutes.

Just out of interest... Why would one want to use THIS instead of just the docker container that we always keep up to date [1] directly :thinking: 

I mean, if you have docker installed, you just need to
```
docker run --rm -v /path/to/fonts:/in:Z -v /path/for/output:/out:Z nerdfonts/patcher [OPTIONS]
```
which will download the most up to date font-patcher etc etc.
Maybe this is a leftover from the time where we did not automagically rebuild the container on every push?

[1] https://hub.docker.com/r/nerdfonts/patcher

**Pull request merged**